src/sentry/api/endpoints/organization_events_trends.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import absolute_import
+
+import sentry_sdk
+
+from datetime import datetime, timedelta
+from rest_framework.response import Response
+from rest_framework.exceptions import ParseError
+
+from sentry import features
+from sentry.api.bases import OrganizationEventsV2EndpointBase, NoProjects
+from sentry.api.event_search import DateArg, parse_function
+from sentry.snuba import discover
+
+
+class OrganizationEventsTrendsEndpoint(OrganizationEventsV2EndpointBase):
+    trend_columns = {
+        "p50": {
+            "format": "percentile_range(transaction.duration, 0.5, {start}, {end}, {index})",
+            "alias": "percentile_range_",
+        },
+        "avg": {
+            "format": "avg_range(transaction.duration, {start}, {end}, {index})",
+            "alias": "avg_range_",
+        },
+        "user_misery": {
+            "format": "user_misery_range({}, {start}, {end}, {index})",
+            "alias": "user_misery_range_",
+        },
+        "count_range": {"format": "count_range({start}, {end}, {index})", "alias": "count_range_"},
+    }
+
+    def has_feature(self, organization, request):
+        return features.has("organizations:internal-catchall", organization, actor=request.user)
+
+    def get(self, request, organization):
+        if not self.has_feature(organization, request):
+            return Response(status=404)
+
+        with sentry_sdk.start_span(op="discover.endpoint", description="filter_params") as span:
+            span.set_tag("organization", organization)
+            try:
+                params = self.get_filter_params(request, organization)
+            except NoProjects:
+                return Response([])
+            params = self.quantize_date_params(request, params)
+
+            has_global_views = features.has(
+                "organizations:global-views", organization, actor=request.user
+            )
+            if not has_global_views and len(params.get("project_id", [])) > 1:
+                raise ParseError(detail="You cannot view events from multiple projects.")
+
+            middle = params["start"] + timedelta(
+                seconds=(params["end"] - params["start"]).total_seconds()
```

Comment:
I think you need to validate intervalRatio here. What if I pass `intervalRatio=0` or `intervalRatio=-1`?

Reply:
Also none of your tests pass in `intervalRatio`, is this parameter used?

Reply:
Ah not yet, thanks for catching this. Was from the even more experimental version of this endpoint. I'll remove it from this PR :+1: